### ISSUE TITLE
[Issue] A regex doesnt match on the app, but does on regex101

### ISSUE THREAD

**Issue by Palloxin** [sky10p/regexh-contributions#30]
Regex101:
![immagine](https://github.com/user-attachments/assets/b7c43f78-c9ac-4931-a5c5-3bb2ec174c35)

App:
![Screenshot_20240728-104903_1](https://github.com/user-attachments/assets/08e14bcd-437d-49dd-899f-49752b4f6505)

regex:
```js
(?<=(?:height|altitude|length|width|wingspan) of (?:almost|over)?|as long as )(?=[aefnost\d])
```
string: ``height of one hundred feet``

I have no idea why it doesnt match, no errors whatsoever showed.

Every flavour in regex101 gives either "error pattern" or a "match". The app doesnt show any error.
Edit: somehow the app doesnt match empty space(position) anymore.

**Comment by sky10p** [comment]
I have copied the regex you wrote in this issue in regex101 with Java 8 and no match is found

![image](https://github.com/user-attachments/assets/83781b44-b644-4454-b574-3808d8393205)

**Comment by Palloxin** [comment]
![immagine](https://github.com/user-attachments/assets/7a3ce208-6d05-44bf-b287-47688f1e6657)

I just copied the regex and string from this issue, it works fine. Maybe try again idk.

**Comment by sky10p** [comment]
I was able to reproduce the issue and found the differences using regex101. In my case, I am discarding all empty strings, while regex101 matches an empty string. Does this really make sense?

Could you explain what the purpose of this regular expression is? I'm analyzing it and can't find any relevant information on matching empty strings. In fact, if I make the changes in my app, you might see one match, but nothing is displayed in the text because it is an empty string.

**Comment by Palloxin** [comment]
I recall the app matched this kind of regexes in previous. It was just missing the purple like in the image above.

**Comment by sky10p** [comment]
This will be fixed in version 5.8.2. It may take some time to be deployed as it is currently under review by Google.
